spec/01-unit/19-hybrid/03-compat_spec.lua

```diff
@@ -494,6 +525,27 @@ describe("kong.clustering.compat", function()
       assert.is_nil(assert(plugins[2]).instance_name)
     end)
 
+    it("plugin.queue_parameters #xxx", function()
+      local has_update, result = compat.update_compatible_payload(config, "3.2.0", "test_")
+      assert.truthy(has_update)
+      result = cjson_decode(inflate_gzip(result)).config_table
+      local plugins = assert(assert(assert(result).plugins))
+      for _, plugin in ipairs(plugins) do
```

Comment:
Direct access to a `plugin` mock object can save you a the iteration here.

Reply:
That would have been nice, but the plugins are actually returned as an array by `update_compatible_payload`, so iteration (or direct index access) are required.